### PR TITLE
Fix Insufficient stock error in checkoutLinesUpdate with 0 quantity.

### PR DESCRIPTION
**Pull request by CossackDex**
I want to merge this change because it fix bug with inappropriate mutation behaviour.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2674 Insufficient stock error on setting quantity to 0](https://app.clickup.com/t/2549495/SALEOR-2674)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/checkout/tests/test_checkout_lines.py [hunk 1]
```diff
@@ -512,6 +512,26 @@ def test_checkout_lines_update_variant_quantity_over_avability_stock(
...
 
 
+def test_checkout_lines_delete_with_by_zero_quantity_when_variant_out_of_stock(
+    user_api_client, checkout_with_item
+):
+    checkout = checkout_with_item
+    line = checkout.lines.first()
+    variant_id = graphene.Node.to_global_id("ProductVariant", line.variant.pk)
+    line.quantity = 0
+    line.save()
```
Comment (IKarbowiak) [inline saleor/graphql/checkout/tests/test_checkout_lines.py:522]: You can use `update_fields` in `save`:
```suggestion
    line.save(update_fields=["quantity"])
```

File: saleor/graphql/checkout/tests/test_checkout_lines.py [hunk 2]
```diff
@@ -512,6 +512,26 @@ def test_checkout_lines_update_variant_quantity_over_avability_stock(
     assert data["checkout"]["lines"][0]["quantity"] == variables["lines"][0]["quantity"]
 
 
+def test_checkout_lines_delete_with_by_zero_quantity_when_variant_out_of_stock(
+    user_api_client, checkout_with_item
+):
+    checkout = checkout_with_item
+    line = checkout.lines.first()
+    variant_id = graphene.Node.to_global_id("ProductVariant", line.variant.pk)
+    line.quantity = 0
```
Comment (korycins) [inline saleor/graphql/checkout/tests/test_checkout_lines.py:521]: Shouldn't be set to something different than 0 here? Assuming that we have some quantity assigned in Checkout. Then we call mutation to set it to 0. 
Also, you should modify the assigned stock to this variant and modify it to 0.

File: saleor/warehouse/availability.py [hunk 3]
```diff
@@ -77,11 +77,12 @@ def check_stock_quantity_bulk(
...
-                    variant=variant, available_quantity=available_quantity
+            if quantity != 0:
+                insufficient_stocks.append(
+                    InsufficientStockData(
+                        variant=variant, available_quantity=available_quantity
+                    )
                 )
-            )
         elif variant.track_inventory:
             if quantity > available_quantity:
```
Comment (korycins) [inline saleor/warehouse/availability.py:87]: And what about this case? Won't trigger the same exception? 
Assuming:
quantity == 0
available_quantity == 0
variant.track_inventory == True

Comment (maarcingebala) [reply saleor/warehouse/availability.py:87]: Maybe we run all checks only when quantity is positive:
```
if quantity > 0:
    if not stocks:
        # raise error
    elif variant.track_inventory and quantity > available_quantity:
       # raise error
```

**Review by korycins — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

**Pull request merged**